src/wallet/wallet.cpp

```diff
@@ -146,6 +146,15 @@ std::shared_ptr<CWallet> LoadWallet(interfaces::Chain& chain, const WalletLocati
     }
     AddWallet(wallet);
     wallet->postInitProcess();
+
+    {
+        // Update wallet transactions with current mempool transactions.
+        auto locked_chain = chain.lock();
+        for (const CTransactionRef& tx : locked_chain->getMemoryPoolTransactions()) {
+            wallet->TransactionAddedToMempool(tx);
```

Comment:
Could there be a race, since we don't take cs_main when removing txs from the mempool?

I.e. a tx is first removed due to eviction, the wallet is notified and then it is re-added here.

Reply:
Indeed, between `getMemoryPoolTransactions` and `TransactionAddedToMempool` a tx can be evicted.

I see 2 solutions:
 - lock the wallet before the loop so that `TransactionRemovedFromMempool` comes after this
 - add a way to lock the mempool in `interfaces::` but not sure if this is wanted? cc @ryanofsky

Reply:
If you move the loop into `postInitProcess`, you can just take a `LOCK(::mempool.cs)` in there

Reply:
@MarcoFalke `postInitProcess` is wallet module, should not use that right? https://github.com/bitcoin/bitcoin/pull/15652#discussion_r268407542

Reply:
Yeah, you'd have to move it to the interface

Alternatively you could add it to `handleNotifications` as suggested by @ryanofsky, but then it would also be called for `createwallet`, which seems a bit wasteful

Reply:
@ryanofsky nice, I thought of something like `replayMemoryPoolNotifications`. However I'm not sure if these notifications should come after `ReacceptWalletTransactions`?

Reply:
re: https://github.com/bitcoin/bitcoin/pull/15652#discussion_r268645607

> Alternatively you could add it to handleNotifications as suggested by @ryanofsky, but then it would also be called for createwallet, which seems a bit wasteful

Agree, this would be too wasteful. Having the separate method to replay notifications is probably better.

> I'm not sure if these notifications should come after ReacceptWalletTransactions?

I'm not 100% sure, but I don't think I see a problem if notifications about what's already in the mempool happen before the wallet adds more transactions to the mempool. I think what you have now basically seems good though.